inlang/source-code/sdk/src/createMessageLintReportsQuery.ts

```diff
@@ -1,50 +1,78 @@
-import { createEffect } from "./reactivity/solid.js"
 import { createSubscribable } from "./loadProject.js"
-import type { InlangProject, InstalledMessageLintRule, MessageLintReportsQueryApi } from "./api.js"
+import type {
+	InlangProject,
+	InstalledMessageLintRule,
+	MessageLintReportsQueryApi,
+	MessageQueryApi,
+} from "./api.js"
 import type { ProjectSettings } from "@inlang/project-settings"
 import type { resolveModules } from "./resolve-modules/index.js"
 import type { MessageLintReport, Message } from "./versionedInterfaces.js"
 import { lintSingleMessage } from "./lint/index.js"
 import { ReactiveMap } from "./reactivity/map.js"
+import { debounce } from "throttle-debounce"
+import { createEffect } from "./reactivity/solid.js"
 
 /**
  * Creates a reactive query API for messages.
  */
 export function createMessageLintReportsQuery(
-	messages: () => Array<Message> | undefined,
+	messagesQuery: MessageQueryApi,
 	settings: () => ProjectSettings,
 	installedMessageLintRules: () => Array<InstalledMessageLintRule>,
-	resolvedModules: () => Awaited<ReturnType<typeof resolveModules>> | undefined
+	resolvedModules: () => Awaited<ReturnType<typeof resolveModules>> | undefined,
+	hasWatcher: boolean
 ): InlangProject["query"]["messageLintReports"] {
 	// @ts-expect-error
 	const index = new ReactiveMap<MessageLintReport["messageId"], MessageLintReport[]>()
 
-	createEffect(() => {
-		const modules = resolvedModules()
-		const _messages = messages()
-		const _settings = settings()
+	const modules = resolvedModules()
 
-		if (_messages && _settings && modules) {
-			// index.clear()
-			for (const message of _messages) {
-				// TODO: only lint changed messages and update arrays selectively
+	const rulesArray = modules?.messageLintRules
+	const messageLintRuleLevels = Object.fromEntries(
+		installedMessageLintRules().map((rule) => [rule.id, rule.level])
+	)
+	const settingsObject = () => {
+		return {
+			...settings(),
+			messageLintRuleLevels,
+		}
+	}
+
+	const messages = messagesQuery.getAll() as Message[]
 
-				lintSingleMessage({
-					rules: modules.messageLintRules,
-					settings: {
-						..._settings,
-						messageLintRuleLevels: Object.fromEntries(
-							installedMessageLintRules().map((rule) => [rule.id, rule.level])
-						),
-					},
-					messages: _messages,
-					message: message,
-				}).then((report) => {
-					if (
-						report.errors.length === 0 &&
-						JSON.stringify(index.get(message.id)) !== JSON.stringify(report.data)
-					) {
-						index.set(message.id, report.data || [])
+	createEffect(() => {
+		if (rulesArray) {
+			for (const messageId of messagesQuery.includedMessageIds()) {
+				messagesQuery.get.subscribe({ where: { id: messageId } }, async (message) => {
```

Comment:
why do you use subscribe? 

you can directly use `get()` without the subscribe callback.

Reply:
![image](https://github.com/inlang/monorepo/assets/58360188/22506a78-7f21-4fe0-ae69-fc8dbcd887ae)

I don't think so. It's not being converted to a signal. It only returns a value this way.

Reply:
haha sdk trapped me :D

Reply:
@samuelstroschein We changed this. It is now a signal within a createEffect.